### ISSUE TITLE
False positive: Missing lifetime lifetime specifier for Self

### ISSUE THREAD

**Issue by torkleyy** [intellij-rust/intellij-rust#1134]
```rust
trait Foo {
    type A;
}

struct Bar<'a>(&'a str);

impl<'a> Foo for Bar<'a> {
    type A = Self;
    // Warning here: "Missing lifetime specifier [E0106]"
}
```

**Comment by alygin** [comment]
What version of the plugin do you use? The problem should have been fixed in the yesterday alpha (0.1.0.1814).

**Comment by alygin** [comment]
Oh, sorry, the fix hasn't made it to alpha yet. But nightly is already patched.

**Comment by torkleyy** [comment]
OK, didn't know
